src/PolykeyAgent.ts

```diff
@@ -61,6 +61,8 @@ type PolykeyAgentOptions = {
     certDuration: number;
     certRenewLeadTime: number;
     recoveryCode: string;
+    privateKeyPath: string;
+    privateKey: PrivateKey;
```

Comment:
The `privateKey: PrivateKey` should be using a more open type, instead the opaque type because of `PolykeyAgent` taking parameters from outside world.

So if you want to keep this then you should be using `privateKey: Buffer`.

Also `privateKey` should be moved up top.

```
privateKey: Buffer;
privateKeyPath: string;
```

Reply:
These 2 options aren't even being used, if these 2 options are to be used, they should be passed to the `KeyRing.createKeyRing`.

Reply:
@addievo link the new PR to fix this.

Reply:
https://github.com/MatrixAI/Polykey/pull/601

Reply:
The `PolykeyAgent.start` should be taking `DeepPartial` of options, and the options need to `keys: { recoveryCode; privateKey; privateKeyPath; }`.

And also in the `KeyRing.createKeyRing` and `KeyRing.start` methods, these need to receive those options if you want to make them useful.

Reply:
Also the `PolykeyAgentOptions` needs to match:

```
  } & ( // eslint-disable-next-line @typescript-eslint/ban-types
    | {}
    | {
        recoveryCode: RecoveryCode;
      }
    | {
        privateKey: PrivateKey;
      }
    | {
        privateKeyPath: string;
      }
```

Which ensures mutually independent options.

Reply:
So:

```
keys: ({} | { recoveryCode: string; } | { privateKey: Buffer } | { privateKeyPath: string })
```

And `PolykeyAgent.start` should take this.